agents/hermes/mcp-config-transaction.py

```
@@ -680,6 +681,26 @@ def apply_transaction_and_reload(
+        # succeeded; rolling it back would undo a live gateway configuration.
+        try:
+            compatibility_hash_path = os.path.join(HERMES_DIR, ".config-hash")
+            integrity = guard.inspect_mcp_integrity_snapshot(
+                HERMES_DIR,
+                STRICT_HASH_PATH if privileged else compatibility_hash_path,
+                compatibility_hash_path if privileged else None,
+            )
+            if integrity.config_text == expected_text and integrity.state == "current":
+                guard.assert_mcp_integrity_snapshot_current(integrity)
+                return {"ok": True, "changed": True, "reloaded": True}
+        except Exception as recovery_error:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)